Sources/System/FileOperations.swift

```diff
@@ -371,3 +371,109 @@ extension FileDescriptor {
     fatalError("Not implemented")
   }
 }
+
+extension FileDescriptor {
+  /// Opens or creates a file for reading or writing.
+  ///
+  /// - Parameters:
+  ///   - path: The location of the file to open.
+  ///   - at: if `path` is relative, treat it as relative to this file descriptor
+  ///     rather than relative to the current working directory
+  ///   - mode: The read and write access to use.
+  ///   - options: The behavior for opening the file.
+  ///   - permissions: The file permissions to use for created files.
+  ///   - retryOnInterrupt: Whether to retry the open operation
+  ///     if it throws ``Errno/interrupted``.
+  ///     The default is `true`.
+  ///     Pass `false` to try only once and throw an error upon interruption.
+  /// - Returns: A file descriptor for the open file
+  ///
+  /// The corresponding C function is `openat`.
+  @_alwaysEmitIntoClient
+  public static func open(
+    _ path: FilePath,
+    at: FileDescriptor,
```

Comment:
```suggestion
    at base: FileDescriptor,
```

(etc, throughout this PR.)